### PR TITLE
Added dependabot automated dependency management

### PR DESCRIPTION
**Pull request by Glennmen**
* Added Dependabot support for automated dependency version management
* Added a Github build action

This will make PRs for new dependency updates, it will not auto merge them. It should make it easier for us to stay on top of dependency updates.
The Github action *should* catch breaking changes in dependency updates. It does a basic Gradle build.
Current config is set to check `daily` for updates, I would recommend this setting until Dependabot has processed all dependencies. After that we can change it to `weekly`.
Github build action will only run on PRs (branches) created by dependabot, but we could extend it to run on all PRs.

For questions about Dependabot or Github actions let me know 😉

### CODE REVIEW

**Review by ChristianoBolla-Appwise — APPROVED** [review]

**Comment by Glennmen** [comment]
@dieter115 Did you have any more questions about the changes?
If not this can be merged I guess.

**Comment by ChristianoBolla-Appwise** [comment]
Maybe one question... So whenever this bot notices an update is available, it will create a PR with the updated version.
We are then supposed to checkout this PR and see if everything still builds without issues. If an issue exists we have to update it within this PR itself... correct? 🤔

**Comment by Glennmen** [comment]
> Maybe one question... So whenever this bot notices an update is available, it will create a PR with the updated version.

That is correct

> We are then supposed to checkout this PR and see if everything still builds without issues. If an issue exists we have to update it within this PR itself... correct? 🤔

I added a build step so when the PR is made a Gradle build in Github actions will also be done so we might be able to catch any issues there without checkout, but it doesn't hurt to double check of course 😉
You can indeed push changes to the PR but you could also make an other PR with the changes, dependabot is smart enough to close the PR it created when it detects that the update has be done with an other PR or direct push.

**Comment by ChristianoBolla-Appwise** [comment]
Okay, great, thanks for the extra information.

**Comment by Glennmen** [comment]
I will merge this after PR #28 has been merged.

**Pull request merged**
